lib/ansible/plugins/connection/psrp.py

```diff
@@ -492,7 +500,8 @@ def _build_kwargs(self):
         else:
             self._psrp_cert_validation = True
 
-        self._psrp_connection_timeout = int(self.get_option('connection_timeout'))
+        self._psrp_connection_timeout = self.get_option('connection_timeout')  # Can be None
```

Comment:
Right now it can't be None, you would have to optionally add it to the kwargs being set as pypsrp cannot handle an explicit None value for this. This is the same as `read_timeout`.

If you are going for the optional route, you also have to remove the `default: ...` entry in the documentation as `get_option` uses this value when it isn't defined.

Reply:
AFAIR None means no connection or read timeout ?

Reply:
Ahh yes sorry I was misreading the intent at the time but can see Ansible will use `30` by default and a user can do `ansible_psrp_connection_timeout:` or `ansible_psrp_connection_timeout: null` to make the value `None`.